common/blocks.py

```diff
@@ -503,7 +503,7 @@ class VerticalBarChart(blocks.StructBlock):
     group_by = blocks.ChoiceBlock(
         label='Group Incidents By',
         required=False,
-        choices=charts.IncidentBranches.choices,
+        choices=charts.IncidentBranchesAndTags,
```

Comment:
I was going to say that I'd kind of prefer the tags be separated into their own field on `VerticalBarChart` instead of being combined here because (1) the editor UX would be nicer to be able to use the autocomplete widget to choose the tag, and (2) because then we could avoid inferring what the user has chosen when we're rendering the chart, _but_ then I remembered that actually we can't use the autocomplete widget here since we're in a `StructBlock`! The other place where we're letting the user choose tags, in the incident set, we have a drop down. So now I'm kind of not feeling as strongly about separating this into another field. @harrislapiroff any thoughts?

Reply:
I think we should separate it out into a separate field in anticipation of using an autocomplete widget in the future (How hard would it be to use an autocomplete widget for a block? That feels like it should definitely be a feature.)

Reply:
In this case, is there a way to limit users to only choosing either a category or a tag? I don't know what the behavior should be if they happened to select group by a category and a tag?

Reply:
Sadly, I don't think wagtail 4.1 supports this kind of strict validation. I'm told it's something that's out there in a future wagtail version (maybe 5+? [Here's some documentation](https://docs.wagtail.org/en/v5.1.2/advanced_topics/streamfield_validation.html)) which we are probably going to upgrade to Soon(tm).

Given this, I'd be fine with adding some help text explaining that these options are mutually exclusive. While at a system level we can't really stop them from choosing both (yet), I think for now it's fine to have one mode "win" if someone picks both a tag and a category. @harrislapiroff thoughts on this?

Reply:
@chigby I agree that's a good approach. We already have places where we do this—for example button and nav fields that accept either a page, document, or URL. Maybe we can open a ticket for adding better validation pending the next version of Wagtail. (Maybe a new label "blocked on:Wagtail 5" or something?)

Reply:
sounds good, let me see what I can do